pkg/server/handler.go

```diff
@@ -534,19 +534,27 @@ func (al *activeListener) OnNewConnection(ctx context.Context, conn api.Connecti
 		return
 	}
 	ac := newActiveConnection(al, conn)
+	// connection may be closed by upstream before added to conn list
+	if conn.State() == api.ConnClosed {
+		return
+	}
+
+	if conn.LocalAddr().Network() == "udp" {
+		network.SetUDPProxyMap(network.GetProxyMapKey(conn.LocalAddr().String(), conn.RemoteAddr().String()), conn)
+	}
 
 	e := al.conns.PushBack(ac)
 	ac.element = e
 
+	// add event listener after setting ac.element to avoid delete empty element
+	ac.conn.AddConnectionEventListener(ac)
```

Comment:
There is another issue that `ac` would not be removed if conn has closed at this time.